dkan_dataset.module

```diff
@@ -247,6 +247,29 @@ function dkan_dataset_get_resource_nodes($nid) {
 }
 
 /**
+ * Helper to get the format string for a resource node
+ *
+ * @param $node
+ *   A resource node
+ *
+ * @return
+ *   The format string (eg, "csv", "html", etc.). Defaults to "data".
+ */
+function dkan_dataset_get_resource_format($node) {
+  $node_wrapper = entity_metadata_wrapper('node', $node);
+  if ($node_wrapper->getBundle() != 'resource') {
+    return FALSE;
+  }
+  if (isset($node->field_format['und']) && $node_wrapper->field_format->value()->name) {
+    return $node_wrapper->field_format->value()->name; 
+  }
+  else {
+
```

Comment:
Oops, that else was to be where a bunch of crazy mimetype discovery logic went but decided not to do that.